gen/list.go

```diff
@@ -187,6 +187,64 @@ func (l *listGenerator) Encode(g Generator, spec *compile.ListSpec) (string, err
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
 
+// Decoder generates a function to read a list of the given type from a
+// stream.Reader.
+//
+//     func $name(sr *stream.Reader) ($listType, error) {
+//             ...
+//     }
+//
+// And returns its name.
+func (l *listGenerator) Decoder(g Generator, spec *compile.ListSpec) (string, error) {
```

Comment:
Call this "Decode"

Reply:
Discussed offline - kept these as `Decoder` since the generation functions in these are in the active voice (e.g., `Reader`).